client/src/pages/Products.js

```diff
@@ -28,27 +28,43 @@ export default function Products() {
   }
   function deleteProduct(id) {
     try {
-      axios.delete('http://localhost:3000/api/merchantproducts/' + id, { crossdomain: true });
+      axios.delete('http://localhost:3000/api/merchantproducts/' + id);
     } catch (error) {
       console.error(error);
     }
   }
-  function updateProduct(url) {
+  function updateProduct(url, id) {
     try {
-      axios.put('http://localhost:3000/api/merchantproducts/', { params: { url: url } });
+      axios.put('http://localhost:3000/api/merchantproducts/', { params: { url: url, id: id } });
     } catch (error) {
       console.error(error);
     }
   }
 
+  async function searchProducts(searchQuery) {
+    const response = await axios.get('http://localhost:3000/api/search/' + searchQuery);
+    setProducts(response.data);
+  }
+
+  function handleChange(value) {
+    setTimeout(() => {
```

Comment:
The timeout delays requests by 5 seconds on search. 
Why? :) 

You still have the same amount of requests as input changes.
Writing "Lampe" results in 5 requests. Take a look at the chrome dev tools network tab.

I guess you like to want only have one request if you write something in your search input?
Take a look at throttle/debounce. You could think about a custom hook too.
Note: You need to clear the old timeout if you create a new timeout.